src/Symfony/Component/Form/Extension/Core/Type/DateTimeType.php

```diff
@@ -126,7 +136,9 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'invalid_message_parameters',
             ]));
 
-            if (isset($emptyData['time'])) {
+            if ($emptyData instanceof \Closure) {
```

Comment:
What about doing the same as what you did in the `DateType` then?

```php
if ($emptyData instanceof \Closure) {
    $lazyEmptyData = static function ($option) use ($emptyData) {
        return static function (FormInterface $form) use ($emptyData, $option) {
            $emptyData = $emptyData($form->getParent());
            return isset($emptyData[$option]) ? $emptyData[$option] : '';
        };
    };
    $dateOptions['empty_data'] = $lazyEmptyData('date');
} else {
    if (isset($emptyData['date'])) {
        $dateOptions['empty_data'] = $emptyData['date'];
    }

    if (isset($emptyData['time'])) {
        $timeOptions['empty_data'] = $emptyData['time'];
    }
}
```

Reply:
To do that the `$timeOptions` initialization should be moved before the first empty data definition.

Reply:
I can see there are conflicting comments from Nicolas and Christian, so maybe we should always set `$lazyEmptyData` base on the instance of once, then test if `$lazyEmptyData` is null to define the options.